### PR TITLE
Output esm, cjs, and umd packages

### LINKED ISSUE #17: Support Consumption of Non-Webpacked Source Code
## Are you reporting a feature request or a bug?
Feature Request

## Request Details
Currently, the only method of consumption is via the webpacked bundle in the dist folder. This webpack bundle includes a polyfill babel polyfill set up via the usage method. This ends up adding a polyfill for Promises but not for the 'finally' functionality of promises since those are not used within the code. One of our solutions relies on a promise polyfill which is a different polyfill than the provided by babel. 

Since the incomplete babel polyfill is loaded first without all the functionality, it doesn't get overwritten by our polypill and thus we end up with the 'finally' functionality missing. As a work around, we've copied the source code into our own project which allows us to use our internal strategy of building projects. This comes with the downside of having to continually update our internal fork over time and maintain that ourselves. It would be ideal for us to be able to consume the non-webpacked source and be able to apply our own polyfills as needed.

### PR DESCRIPTION
**Pull request by MMMalik**
### Changes

- replaced `webpack` with `rollup`
- configured `rollup` so that 3 packages are created: `cjs`, `esm`, `umd`
- copied script from seemingly abandoned `load-script` package to a local file
- updated `package.json` with `main` and `module` fields, removed `load-script` from dependencies

### Package size changes

| Package | Size changes |
| --- | --- |
| ckeditor4-integrations-common | umd: **21.3 KB** -> **1.9 KB** |
| ckeditor4-react | v1 umd: **51.7 KB** -> **39.9 KB**<br> v2 umd: **31.1 KB** -> **11.5 KB**  |
| ckeditor4-angular | sample `ng` app: **167.3 KB** -> **147.4 KB** |
| ckeditor4-vue | `ckeditor` umd: **24.2 KB** ->  **4.3 KB**<br> `legacy` umd: **35.7 KB** -> **26.7 KB** |

* Package weighs around ~20 KB less if no polyfills are included (neither in `ckeditor4-integrations-common` nor downstream)
* Package weighs around ~10 KB less if IE11 polyfills are provided downstream (React v1, Vue `legacy` package)

### Further steps

1. I believe this is a BREAKING CHANGE for this library. I suggest to release these changes as `1.0.0`.

2. `ckeditor4-react` v1 can be adjusted easily to use `1.0.0` (so no breaking change for `ckeditor4-react` v1). `ckeditor4-react` v2 must use `1.0.0` before its release.

3. Adjusting `ckeditor4-vue` to use `1.0.0` is trivial but is also a potentially BREAKING CHANGE because some users might rely on polyfilled version of `ckeditor4-integrations-common`. I would risk releasing a `minor` version though because `ckeditor4-vue` makes **clear distinction** between so-called `ckeditor` es6-compatible package and a `legacy` es5-compatible one.

4. Adjusting `ckeditor4-angular` to use `1.0.0` is trivial but is also a potentially BREAKING CHANGE because some users might rely on polyfilled version of `ckeditor4-integrations-common`. I would NOT risk releasing a minor version here since `ckeditor4-integrations-common` is treated as external dependency in Angular integration.

### To test

All integrations have been tested by referencing [this commit](https://github.com/MMMalik/ckeditor4-integrations-common-t-17/tree/52b4f73b655edebcfc30d8639f98633214997fe9).

Example flow:

1. Go to local clone of `ckeditor4-react`
2. Add to dependencies: `"ckeditor4-integrations-common": "git://github.com/mmmalik/ckeditor4-integrations-common-t-17.git#1230f21790ff03bb2f44fea40de8fcb491952f5d"`
3. Install deps: `npm i`
3. Run build: `npm run build`
4. Run samples: `http-server -s -o samples/index.html`
5. ~Optional: to make IE11 work adjust `babel-loader` in `webpack.config.js` with `/node_modules\/(?!(ckeditor4-integrations-common)\/).*/`~ Code is compatible with ES5. There is no need to transpile it. Only polyfills must be included for older browsers.

Similar flow applies to `ckeditor4-vue`. 

In order to test `ckeditor4-angular` create new sample app with `ng new ...`. Then reference `"ckeditor4-angular": "git://github.com/mmmalik/ckeditor4-angular-t-191.git#05a167ffa518337086fb6ee5e38d3279c2bb3a9a"` in dependencies. Follow [this guide](https://github.com/ckeditor/ckeditor4-angular#usage) to run CKEditor.

Closes #17.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,9 +48,11 @@
...
 		"karma-webpack": "^4.0.2",
 		"mocha": "^8.1.3",
+		"rollup": "^2.47.0",
+		"rollup-plugin-cleanup": "^3.2.1",
+		"rollup-plugin-terser": "^7.0.2",
 		"sinon": "^9.0.3",
 		"sinon-chai": "^3.5.0",
-		"webpack": "^4.44.2",
-		"webpack-cli": "^3.3.12"
+		"webpack": "^4.44.2"
```
Comment (Comandeer) [inline package.json:55]: Shouldn't it be removed? :thinking:

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import commonjs from '@rollup/plugin-commonjs';
+import { terser } from 'rollup-plugin-terser';
+import cleanup from 'rollup-plugin-cleanup';
+import { babel } from '@rollup/plugin-babel';
+
+const input = 'src/index.js';
+const banner = `/**
+* @license Copyright (c) 2003-2021, CKSource - Frederico Knabben. All rights reserved.
+* For licensing, see LICENSE.md.
+*/`;
```
Comment (Comandeer) [inline rollup.config.js]:
```suggestion
 * @license Copyright (c) 2003-2021, CKSource - Frederico Knabben. All rights reserved.
 * For licensing, see LICENSE.md.
 */`;
```

File: src/loadScript.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
+/**
+ * The script below was copied over from https://github.com/eldargab/load-script, because the repo seems to be not maintained anymore.
+ *
+ * Version at commit: https://github.com/eldargab/load-script/blob/318ead6c590d403e0fbd4ae6e76b0407614addec/index.js
+ */
+
+module.exports = function load( src, opts, cb ) {
```
Comment (Comandeer) [inline src/loadScript.js]: I'd change it to ESM as it's the only source file now that still uses CJS. And, if I'm not mistaken, this rewrite would also allow us to drop `@rollup/plugin-commonjs`.

**Review by Comandeer — CHANGES_REQUESTED** [review]
Overally looks good, there are some minor issues.

File: package.json (hunk 1 above)
Comment (MMMalik) [reply package.json:55]: Webpack is still used in `karma.conf.js`. It should be replaced with Rollup but I didn't want to extend the scope of this task. Would you agree to create a follow-up task for that?

File: src/loadScript.js (hunk 3 above)
Comment (MMMalik) [reply src/loadScript.js]: That's a good idea. I'll give it a try.

**Comment by MMMalik** [comment]
Thanks @Comandeer. As per your suggestion `loadScript` was transformed from CommonJS into ESM. `@rollup/plugin-commonjs` was removed.

File: package.json (hunk 1 above)
Comment (Comandeer) [reply package.json:55]: Yes, it sounds like a good candidate for a follow-up task.

**Review by Comandeer — APPROVED** [review]
LGTM! 🎉 
Tested with https://github.com/MMMalik/ckeditor4-integrations-common-t-17/tree/52b4f73b655edebcfc30d8639f98633214997fe9 and https://github.com/Comandeer/ckeditor4-angular-t-191/tree/b2be9d3c3054eed10c73694cd833e7c51a2e9d97.

**Pull request merged**